block/block_service.proto

```diff
@@ -0,0 +1,710 @@
+    rpc serverStatus(ServerStatusRequest) returns (ServerStatusResponse);
+
+    /**
+     * Read a single block from the block node.
+     * <p>
+     * The request SHALL describe the block number of the block to retrieve.
+     */
+    rpc singleBlock(SingleBlockRequest) returns (SingleBlockResponse);
+
+    /**
+     * Read a state snapshot from the block node.
+     * <p>
+     * The request SHALL describe the last block number present in the
+     * snapshot.<br/>
+     * Block node implementations MAY decline a request for a snapshot older
+     * than the latest available, but MUST clearly document this behavior.
+     */
+    rpc stateSnapshot(StateSnapshotRequest) returns (StateSnapshotResponse);
+
+    /**
+     * Publish a stream of blocks.
+     * <p>
+     * Each item in the stream MUST contain one `BlockItem`.<br/>
+     * Each Block MUST begin with a single `BlockHeader` block item.<br/>
+     * The block node SHALL append each `BlockItem` to an internal structure
+     * to construct full blocks.<br/>
+     * Each Block MUST end with a single `BlockStateProof` block item.<br/>
+     * It is RECOMMENDED that the implementations verify the Block using the
+     * `BlockStateProof` to validate all data was received correctly.<br/>
+     * This API SHOULD, generally, be restricted based on mTLS authentication
+     * to a limited set of source (i.e. consensus node) systems.
+     */
+    rpc publishBlockStream (stream PublishStreamRequest) returns (stream PublishStreamResponse);
+
+    /**
+     * Subscribe to a stream of blocks.
+     * <p>
+     * Each item in the stream SHALL contain one `BlockItem` or a
+     * response code.<br/>
+     * The request message MUST specify start and end block numbers
+     * to return/<br/>
+     * The block node SHALL stream the full contents of the blocks
+     * requested.<br/>
+     * The block items SHALL be streamed in order originally produced within
+     * a block.<br/>
+     * The blocks shall be streamed in ascending order by `block_number`.<br/>
+     * The block node SHALL end the stream when the last requested block has
+     * been sent.<br/>
+     * The block node SHALL end the stream with response code containing a
+     * status of SUCCESS when the stream is complete.<br/>
+     * The block node SHALL end the stream with a response code containing a
+     * status of `READ_STREAM_INVALID_START_BLOCK_NUMBER` if the start block
+     * number is greater than the end block number.<br/>
+     * The block node SHALL end the stream with a response code containing a
+     * status of `READ_STREAM_PAYMENT_INSUFFICIENT` if insufficient payment
+     * remains to complete the requested stream.<br/>
+     * The block node SHALL make every reasonable effort to fulfill as much
+     * of the request as possible in the event payment is not sufficient to
+     * complete the request.
+     */
+    rpc subscribeBlockStream(stream SubscribeStreamRequest) returns (stream SubscribeStreamResponse);
```

Comment:
Q: why a bi-directional stream for the subscribe?
Would a client make a single request and receive the stream, why would it keep sending a `SubscribeStreamRequest` if blocks are delivered in order?

Reply:
The subscriber is receiving `BlockItem`s, not `Block`s and, though design is not complete for this, will likely need to acknowledge items as they're received.

Streams aren't sent all at once, and can last a rather long (perhaps hours) time.